File: Python/BOJ/1874.py
import sys


def solution():
    n = int(sys.stdin.readline())
    stack = []
    cur = 1
    flag = 0
    answer = []
    for i in range(n):
        num = int(sys.stdin.readline())
        while cur <= num:
            stack.append(cur)
            answer.append("+")
            cur += 1
        if stack[-1] == num:
            stack.pop()
            answer.append("-")
        else:
            print("NO")
            flag = 1
            break

    if flag ==0:
        for i in answer:
            print(i)
solution()

File: C++/BOJ/1874.cpp
#include<iostream>
#include<stack>
#include<vector>

using namespace std;

int main()
{
    stack<int> s;
    vector<char> answer;
    int cur = 1;
    int n;
    cin>>n;

    for (int i = 0; i < n; i++)
    {
        int num;
        cin>>num;

        while (cur <= num)
        {
            s.push(cur);
            cur+=1;
            answer.push_back('+');
        }
        if (s.top() == num)
        {
            s.pop();
            answer.push_back('-');
        }
        else
        {
            cout << "NO";
            return 0;
        }
    }
    for (int i = 0; i < answer.size(); i++)
        cout << answer[i] <<'\n';   
}